man/machine-id.xml

```diff
@@ -115,6 +116,34 @@
     early boot but become writable later on.</para>
   </refsect1>
 
+  <refsect1>
+    <title>First Boot Semantics</title>
+
+    <para><filename>/etc/machine-id</filename> is used to decide whether a boot is the first one.  The rules
+    are as follows:</para>
+
+    <orderedlist>
+      <listitem><para>If <filename>/etc/machine-id</filename> does not exist, this is a first boot.  During
+      early boot, systemd will create it with the content <literal>uninitialized</literal> and bind-mount a
+      temporary file over it which contains the actual machine-id.  Later (after <filename>first-boot.target</filename>
+      was reached), the machine-id will be written to disk.</para></listitem>
+
+      <listitem><para>If <filename>/etc/machine-id</filename> contains the string <literal>uninitialized</literal>,
+      a boot is also considered the first boot.  The same mechanism as above applies.</para></listitem>
+
+      <listitem><para>If <filename>/etc/machine-id</filename> exists and is empty, a boot is
+      <emphasis>not</emphasis> considered the first boot.  systemd will still bind-mount a file containing
+      the actual machine-id over it and later try to commit it to disk (if <filename>/etc</filename> is
+      writable).</para></listitem>
+
+      <listitem><para>If <filename>/etc/machine-id</filename> already contains a valid machine-id, this is
+      not a first boot.</para></listitem>
```

Comment:
Hmm, does this preserve backwards compatibility or is it a change in behaviour?

Reply:
this is how it always was